packages/issuer/lib/CredentialSupportedV1_11Builder.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  CredentialFormat,
+  ICredentialDisplay,
+  ICredentialSupportedV1_11,
+  IIssuerCredentialSubjectDisplayNameAndLocale,
+  IIssuerCredentialSubjectV1_11,
+  invalid_request,
+} from '@sphereon/openid4vci-common'
+
+export class CredentialSupportedV1_11Builder {
+  format?: CredentialFormat
+  id?: string
+  types?: string[]
+  cryptographicBindingMethodsSupported?: ('jwk' | 'cose_key' | 'did' | string)[]
+  cryptographicSuitesSupported?: ('jwt_vc' | 'ldp_vc' | string)[]
+  display?: ICredentialDisplay[]
+  credentialSubject?: IIssuerCredentialSubjectV1_11
+
+  withFormat(credentialFormat: CredentialFormat): CredentialSupportedV1_11Builder {
+    this.format = credentialFormat
+    return this
+  }
+
+  withId(id: string): CredentialSupportedV1_11Builder {
+    this.id = id
+    return this
+  }
+
+  withTypes(type: string | string[]): CredentialSupportedV1_11Builder {
+    if (!Array.isArray(type)) {
+      this.types = this.types ? [...this.types, type] : [type]
+    } else {
+      this.cryptographicBindingMethodsSupported = this.cryptographicBindingMethodsSupported
+        ? [...this.cryptographicBindingMethodsSupported, ...type]
+        : type
+    }
+    return this
+  }
+  withCryptographicBindingMethod(method: string | string[]): CredentialSupportedV1_11Builder {
+    if (!Array.isArray(method)) {
```

Comment:
If method is not an array and cryptographicBindingSupported is not present you return method as an array? Isn't it the other way around?

Reply:
no, it's array in the spec, so I get a single or a list, and I create an array out of it

Reply:
Ok!